tests/test_io.py

```diff
@@ -1690,3 +1691,378 @@ def test_csv_options_cs2(self):
+
+        fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+        result_handler.integration_point()
+
+        fmu.time += 0.01
+        fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+
+        rdb = ResultDymolaBinary(opts["result_file_name"], allow_file_updates=True)
+        nbr_of_calls = nbr_of_calls
+        diag_data_ratio = diag_data_ratio
+        counter = 0
+        start_index = 0
+
+        nbr_of_points = 5
+        h = 1/nbr_of_points * 0.1 # "arbitrary" step through time
+
+        data_to_return = {}
+        for i in range(nbr_of_calls):
+            time.sleep(0.1) # without this the tests fail on Linux, should look into a proper way of solving this
+            for j in range(nbr_of_points):
+                result_handler.integration_point()
+                if opts["dynamic_diagnostics"] and counter%diag_data_ratio==0:
+                    diag_vars['@Diagnostics.step_time'] = (h, 'Step time') # arbitrary test value
+                    result_handler.diagnostics_point(np.array([val[0] for val in diag_vars.values()], dtype=float))
+
+                counter += 1
+                fmu.time += h
+                fmu.set('J4.phi', math.sin(5*fmu.time + math.pi/2)) # arbitrary
+
+            trajectories, start_index = rdb.get_variables_data(vars_to_test, start_index, u(start_index))
+            data_to_return[i] = [t.x for t in trajectories]
+
+        assert data_to_return, "Something went wrong, no test data was generated"
+        return data_to_return
+
+    @testattr(stddist = True)
+    def test_get_variables_data_values0(self):
+        """ Verifing values from get_variables_data. """
+        vars_to_test = ['J4.phi']
+        test_data_sets = self._test_get_variables_data(False, 3, None, vars_to_test, lambda x: None, "TestFile00.mat")
+
+        reference_data = {
+            0: [1.00000000, 0.99875026, 0.98877108, 0.96891242, 0.93937271, 0.90044710],
+            1: [0.85252452, 0.79608380, 0.73168887, 0.65998315, 0.58168309],
+            2: [0.49757105, 0.40848744, 0.31532236, 0.21900669, 0.12050277]
+        }
+
+        for index, test_data in test_data_sets.items():
+            np.testing.assert_array_almost_equal(test_data[0], reference_data[index])
+
+    @testattr(stddist = True)
+    def test_get_variables_data_values1(self):
+        """ Verifing values from get_variables_data, with dynamic_diagnostics = True. """
+        vars_to_test = ['time', 'J4.phi', '@Diagnostics.step_time', '@Diagnostics.nbr_steps']
+        test_data_sets = self._test_get_variables_data(True, 5, 3, vars_to_test, lambda x: None, "TestFile01.mat")
+
+        reference_data = {
+            0: [ 1.00000000,  0.99875026,  0.93937271],
+            1: [ 0.7960838,   0.58168309],
+            2: [ 0.31532236],
+            3: [ 0.02079483, -0.27559025],
+            4: [-0.54735767, -0.77023125]
+        }
+
+        # Just verify results for J4.phi here, but we retrieve all four trajectories at once
```

Comment:
Do we still need the loop below then? 
The code appears more complicated than it should be here?

Reply:
The loop we need in order to test data from all 5 calls, since we loop over "sets of generated data".
Note also we add all of those variables to the call just to make sure the data at least for 1 of the variables (`J4.phi`) is as expected when asking for multiple variables at once.